backend/services/ffmpeg_utils.py

```
@@ -104,13 +113,66 @@ def find_ffprobe():
+    proc = await loop.run_in_executor(None, _run)
+    # Wrap the Popen process to match asyncio.subprocess.Process interface
+    class _AsyncCompatProc:
+        def __init__(self, popen):
+            self._popen = popen
+            self.returncode = popen.returncode
+            self.stdin = popen.stdin
+            self.stdout = popen.stdout
+            self.stderr = popen.stderr
+            self.pid = popen.pid
+
+        async def communicate(self, input=None):
```

Ruff (0.15.14)

[error] Function argument `input` is shadowing a Python builtin

(A002)